command/server/config.go

```diff
@@ -151,6 +151,67 @@ ui = true
 	return parsed, nil
 }
 
+// DevTLSConfig is a Config that is used for dev tls mode of Vault.
+func DevTLSConfig(storageType string) (*Config, string, error) {
+	dir, err := os.MkdirTemp("", "vault-tls")
```

Comment:
Maybe `-dev-tls` should take a directory path? AFAICT from `server.go`, this directory gets leaked, per docs:

> It is the caller's responsibility to remove the directory when it is no longer needed. 

...which means we'd either need to clean it up or hope there's plenty of space (there should be).

But, for automation, it'd be great if we could specify the directory ahead of time that we'd expect to use and either have Vault copy the CA there, or just generate everything in that directory (and ignore the perms issue in the other comment).

My 2c. :-)

Reply:
This is a good catch on `os.MkdirTemp`. I'm looking into clean up on that directory now.

Good point about automation and having a configurable for cert directory. I will add it!

Reply:
Yeah, I mean its probably what, 20KB max size? Not like we're leaking lots of data, but if its disk-backed with an older FS, we would be leaking lots of inodes (4? on ext2/ext3?) and that'd warrant the concern more, tbh.

Reply:
Added directory flag (`-dev-tls-cert-dir`) and clean up on shutdown.